### ISSUE TITLE
browserSync task backward compatibility is broken

### ISSUE THREAD

**Issue by maxlk** [lookfirst/gulp-helpers#22]
In gulp-helpers **2.0.17** backward compatibility of **browserSync** task was broken.
Specifically there are two things which break BWC:
1. Now `historyApiFallback` option must be array or string, it's not enough to just specify `historyApiFallback: true`. So, even [systemjs-seed](https://github.com/lookfirst/systemjs-seed) example doesn't work now. It would be nice to return previous default behavior.
2. Path existence is checked without respect of `config.server.routes` option. It means that any custom path mapping is broken. For instance in the following example index.html would be returned instead of css and javascript files:

``` javascript
taskMaker.defineTask('browserSync', {
    taskName: 'serve',
    config: {
        server: {
            baseDir: ['./dist'],
            routes: {
                '/system.config.js': './system.config.js',
                '/jspm_packages': './jspm_packages',
                '/style': './scss/output'
            }
        }
    },
    historyApiFallback: __dirname + '/dist/index.html'
})
```

**Comment by maxlk** [comment]
The problem was caused by this change: [https://github.com/lookfirst/gulp-helpers/pull/21]

**Comment by zivl** [comment]
@lookfirst do you think PR #21 should be reverted (until @cbas will submit his complete solution)?
otherwise we must change major version to prevent BWC issue.

**Comment by lookfirst** [comment]
@zivl Whatever you think is best. =)

**Comment by zivl** [comment]
ok I have an idea which sasisfied all sides + will give @cbas some space to come up with a better solution. I'm bringing back `connect-history-api-fallback` for boolean cases - this will cover BWC + leave @cbas options if stated otherwise

**Comment by zivl** [comment]
fixed in PR #23

**Comment by maxlk** [comment]
Great! Thanks!
